### PR TITLE
computing attention scores using relative attention bias

### PR DESCRIPTION
**Pull request by pmabbo13**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* (to be filled)

### CODE REVIEW

File: torchtext/prototype/t5/modules.py [hunk 1]
```diff
@@ -64,6 +65,65 @@ def __init__(
     def forward():
         pass
 
+    def _t5_scaled_dot_product_attention(
```
Comment (Nayef211) [inline torchtext/prototype/t5/modules.py:68]: Can we leave a comment with the link to the original implementation?

**Review by Nayef211 — APPROVED** [review]

**Comment by pmabbo13** [comment]
# Description

Having computed the relative attention bias term, this method computes the attention scores.  The implementation is very similar to the [nn.Functional._scaled_dot_product_attention](https://github.com/pytorch/pytorch/blob/5953fd9133c0bdcc0158acf1472fac403bc5f636/torch/nn/functional.py#L4814), expect that we pass in `position_bias` as an input argument so that relative attention bias can be incorporated in the computation of the attention scores.

Since the input tensors to this function are 4-dimensional, we replace the `torch.baddbmm`  and `torch.bmm` with `torch.matmul`. Since we are no longer using `torch.baddbmm` where `attn_mask` was passed as an input argument, we instead add the `attn_mask` directly to `position_bias` to ensure the mask is still applied before the softmax is taken to get the final attention scores.
